django/template/loader_tags.py

```diff
@@ -130,6 +130,7 @@ def get_parent(self, context):
 
     def render(self, context):
         compiled_parent = self.get_parent(context)
+        context.render_context.template = compiled_parent
```

Comment:
I want to say this is okay, but it might introduce edge cases since `render_context.template` isn't set back to it's original value. I can't identify a specific issue, but it might be safer to mirror the behavior of `RenderContext.push_state`, minus the calls to `self.push()` and `self.pop()`:

https://github.com/timgraham/django/blob/3c4944bec3935342871ced38242ae1492b5504d3/django/template/context.py#L206